### PR TITLE
Lower LR + longer T_max co-sweep across TF/AF/TFP

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

LR and cosine period (T_max) form a coupled frontier: lower LR explores more carefully but needs longer cosine periods to reach deep basins. The current champion configs were tuned for LR at a fixed T_max (or vice versa), but the joint (LR, T_max) space hasn't been explored below the champion LR for TF, AF, or TFP.

**Evidence:**
- AF breakthrough: T_max=50 at lr=6e-4 gave 19.4% improvement over T_max=10 — longer periods were the dominant lever
- TF progression: lr went 3e-4→2e-4→1.5e-4→1.25e-4, each step improving, but T_max stayed at 10 throughout — the LR descent may not have bottomed out
- TFP has only been tested at one LR (5e-4) and one T_max (150) — unexplored territory
- DM LR is being covered by piccolo #3047, so we skip DM here

The hypothesis: **0.5x–0.8x of champion LR with 1.3x–2x longer T_max finds deeper basins on TF, AF, and TFP.**

## Instructions

Run 8 experiments across 8 GPUs. Use `--wandb_group casca-lr-tmax-cosweep` for all runs.

### TF Runs (2 GPUs) — Lion optimizer, all else matches champion

**Run 1 — TF lr=1e-4, T_max=12 (0.8x LR, 1.2x T_max):**
```bash
python train.py --dataset tandemfoil --optimizer lion --lr 1e-4 \
  --cosine-t-max 12 --grad-clip 0.5 --weight-decay 1e-2 \
  --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 --wandb_group casca-lr-tmax-cosweep
```

**Run 2 — TF lr=7.5e-5, T_max=17 (0.6x LR, 1.7x T_max):**
```bash
python train.py --dataset tandemfoil --optimizer lion --lr 7.5e-5 \
  --cosine-t-max 17 --grad-clip 0.5 --weight-decay 1e-2 \
  --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 --wandb_group casca-lr-tmax-cosweep
```

### AF Runs (3 GPUs) — AdamW optimizer, all else matches champion

**Run 3 — AF lr=5e-4, T_max=50 (0.83x LR, same T_max — isolates LR effect):**
```bash
python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 5e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

**Run 4 — AF lr=4.5e-4, T_max=67 (0.75x LR, 1.34x T_max):**
```bash
python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 4.5e-4 --cosine-t-max 67 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

**Run 5 — AF lr=3e-4, T_max=100 (0.5x LR, 2x T_max — extreme test):**
```bash
python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 3e-4 --cosine-t-max 100 \
  --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

### TFP Runs (3 GPUs) — AdamW optimizer, champion architecture

**Run 6 — TFP lr=3.75e-4, T_max=200 (0.75x LR, 1.33x T_max):**
```bash
python train.py --dataset tandemfoil_paper \
  --optimizer adamw --lr 3.75e-4 --cosine-t-max 200 --no-use-ema \
  --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

**Run 7 — TFP lr=2.5e-4, T_max=300 (0.5x LR, 2x T_max):**
```bash
python train.py --dataset tandemfoil_paper \
  --optimizer adamw --lr 2.5e-4 --cosine-t-max 300 --no-use-ema \
  --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

**Run 8 — TFP lr=5e-4, T_max=50 (same LR, AF's breakthrough T_max — tests T_max transfer):**
```bash
python train.py --dataset tandemfoil_paper \
  --optimizer adamw --lr 5e-4 --cosine-t-max 50 --no-use-ema \
  --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --epochs 999 --wandb_group casca-lr-tmax-cosweep
```

## Design Summary

| Run | Dataset | LR | T_max | vs Champion (LR × T_max) |
|---|---|---|---|---|
| 1 | TF | Lion 1e-4 | 12 | 0.8x LR, 1.2x T |
| 2 | TF | Lion 7.5e-5 | 17 | 0.6x LR, 1.7x T |
| 3 | AF | AdamW 5e-4 | 50 | 0.83x LR, same T |
| 4 | AF | AdamW 4.5e-4 | 67 | 0.75x LR, 1.34x T |
| 5 | AF | AdamW 3e-4 | 100 | 0.5x LR, 2x T |
| 6 | TFP | AdamW 3.75e-4 | 200 | 0.75x LR, 1.33x T |
| 7 | TFP | AdamW 2.5e-4 | 300 | 0.5x LR, 2x T |
| 8 | TFP | AdamW 5e-4 | 50 | same LR, AF's T_max (transfer test) |

## Reporting

Report the primary metric for each dataset: `val_primary/surface_pressure_mae` (TF), `val_primary/surface_mse` (AF), `val_primary/field_mse` (TFP). Include W&B run IDs, best epoch, and final epoch.

## Baseline

- **TandemFoil:** `val_primary/surface_pressure_mae` = **22.537** (#2924 — Lion lr=1.25e-4, T_max=10, gc=0.5, EMA=0.999, 3L/192d)
- **AirfRANS:** `val_primary/surface_mse` = **0.000482** (#2951 — AdamW lr=6e-4, T_max=50, 2L/256d)
- **TandemFoil Paper:** `val_primary/field_mse` = **0.00434** (#2979 — AdamW lr=5e-4, T_max=150, 3L/192d)
- DM skipped — piccolo #3047 covers DM LR fine-tuning

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
